plugin/trino-hive/src/test/java/io/trino/plugin/hive/metastore/glue/TestHiveGlueMetastoreAccessOperations.java

```diff
@@ -87,6 +95,40 @@ public void cleanUpSchema()
         getQueryRunner().execute("DROP SCHEMA " + testSchema + " CASCADE");
     }
 
+    @Test
+    void testInsertOverwriteStatisticsDisabled()
+    {
+        String tableName = "test_insert_overwrite_" + randomNameSuffix();
+
+        try {
+            assertUpdate("CREATE TABLE " + tableName + " (id INT, part INT) WITH (partitioned_by = ARRAY['part'])");
+            assertUpdate("INSERT INTO " + tableName + " VALUES (1, 1), (2, 1)", 2);
+
+            Session insertOverwriteSession = Session.builder(getQueryRunner().getDefaultSession())
+                    .setCatalogSessionProperty("hive", "insert_existing_partitions_behavior", "OVERWRITE")
+                    .setCatalogSessionProperty("hive", "collect_column_statistics_on_write", "false")
+                    .setCatalogSessionProperty("hive", "statistics_enabled", "false")
+                    .build();
+            assertInvocations(insertOverwriteSession, "INSERT INTO " + tableName + " VALUES (3, 1)",
+                    ImmutableMultiset.<GlueMetastoreMethod>builder()
+                            .add(DELETE_COLUMN_STATISTICS_FOR_PARTITION)
+                            .add(GET_COLUMN_STATISTICS_FOR_PARTITION)
+                            .add(GET_PARTITION)
+                            .add(GET_TABLE)
+                            .add(UPDATE_PARTITION)
+                            .build(),
+                    // We can't disable partition cache in glue v2, see GlueMetastoreModule#createGlueCache
+                    // there is maybe 1 time additional call for the GET_PARTITION
+                    ImmutableMultiset.<GlueMetastoreMethod>builder()
+                            .add(GET_PARTITION)
+                            .build());
```

Comment:
Will the invocations be deterministic if we execute `flush_metadata_cache` procedure beforehand?